### PR TITLE
docs: improvements to the README.md

### PR DESCRIPTION
**Pull request by RomainMuller**
Running through the `README.md` file to dust if off and try to improve
it ever so slightly.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: docs/assembly.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+-------------|-------------|----------------------------------------------------
+`kind`       |`'inteface'` |Discriminator to identify interfaces
+`assembly`   |`string`     |The name of the assembly this interface is a part of
+`fqn`        |`string`     |The fully-qualified name of the interface
+`interfaces` |`string[]`   |The fully-qualified names of interfaces extended by this inteface
+`methods`    |`Method[]`   |The [methods] declared by this interface
+`name`       |`string`     |The simple name of the interface
+`properties` |`Property[]` |The [properties] declared by this interface
+
+#### Structures (a.k.a. Data Types)
```
Comment (eladb) [inline docs/assembly.md]: This should be `structs` not `structures`

File: docs/assembly.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+Attribute    | Type         | Description
+-------------|--------------|---------------------------------------------------
+`name`       |`string`      |The name of the enum member. Must be `UPPER_SNAKE_CASED`
+
+#### Methods
+
+Attribute    | Type          | Description
+-------------|---------------|--------------------------------------------------
+`abstract`   |`boolean`      |Whether this method is `abstract`
+`async`      |`boolean`      |Whethert his method is asyncrhonous
```
Comment (eladb) [inline docs/assembly.md]: typo

File: docs/assembly.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+Attribute    | Type         | Description
+-------------|--------------|---------------------------------------------------
+`name`       |`string`      |The name of the enum member. Must be `UPPER_SNAKE_CASED`
+
+#### Methods
+
+Attribute    | Type          | Description
+-------------|---------------|--------------------------------------------------
+`abstract`   |`boolean`      |Whether this method is `abstract`
```
Comment (eladb) [inline docs/assembly.md]: I think the descriptions are a bit lacking. This is where we need to define what each attribute means in terms of design and modeling. For example, `abstract`..

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+package managers, certain fields that are optional in the standard
+[package.json schema] are required by `jsii`.
+
+For example, Maven Central requires packages to carry [sufficient metadata],
+such as *developer information* and *license*, in order to be valid for
+publishing.
+
+Field       | Required | Extensions
+------------|----------|--------------------------------------------------------
+`author`    | Required | `author.organization`
```
Comment (eladb) [inline docs/configuration.md]: Add a `Description` column and explain what this field means or link to doc

**Comment by eladb** [comment]
This is 👍

File: docs/assembly.md (hunk 1 above)
Comment (RomainMuller) [reply docs/assembly.md]: Yeah I tried to take a stance against abbreviating here... How strongly do you feel about that?  Like - do you feel the common use here is sufficient that "structs" conveys additional meaning than "structures"?

Comment (RomainMuller) [reply docs/assembly.md]: Oh well never mind, I changed.

File: docs/assembly.md (hunk 3 above)
Comment (RomainMuller) [reply docs/assembly.md]: I know. I wanna add more detailed verbiage, this is intended to be a high-level (birds-eye) view. I just want to ship this now - it's better than nothing.

File: docs/configuration.md (hunk 4 above)
Comment (RomainMuller) [reply docs/configuration.md]: I agree. I was trying to find the best format for this doc but I was spending too much time. I'll revisit later (probably soon).

**Pull request merged**